src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/Documents/EditorDocumentManagerListener.cs

```diff
@@ -53,6 +64,9 @@ internal partial class EditorDocumentManagerListener : IRazorStartupService
         _joinableTaskContext = joinableTaskContext;
         _telemetryReporter = telemetryReporter;
 
+        _disposeTokenSource = new();
+        _workQueue = new AsyncBatchingWorkQueue<Work>(s_delay, ProcessBatchAsync, _disposeTokenSource.Token);
```

Comment:
>AsyncBatchingWorkQueue [](http://example.com/codeflow?start=25&length=22)

Just to make sure I understand this right - we don't need the comparer for the queue work items because we always want to process all events? This just gives us ability to batch them?

Reply:
Yep, the comparer is just used to de-dupe, but this class just processes every notification. We just want the batching here so that we can process multiple events on a single UI thread switch.